01/highest-rated.md

```diff
@@ -0,0 +1,189 @@
+##### !hint
+
+Consider the following for inspiration:
+
+- [About PSEs](../about-pses/about-pses.md)
+- [Our example PSE with example answers](../about-pses/example-pse.md)
+
+##### !end-hint
+##### !explanation 
+
+Here are some example clarifying questions:
+
+1. What should the function return if there are multiple restaurants tied for the highest rating?
+1. How should the function handle restaurant dictionaries with missing data (name or rating)?
+1. How should the function handle restaurant dictionaries with keys in addition to `"name"` and `"rating"`?
+1. How should the function handle non-numerical values for `"rating"`?
+1. How should the function handle input that is a single dictionary (not a list)?
+ 
+##### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 2 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: 890d7045-28cb-4832-812b-d65803ce2618
+* title: Write Unit Tests
+* topics: pse
+##### !question
+
+1. Use the comments provided to write at least two example input/outputs:
+    * Consider at least one nominal and one edge case.
+    * What is the expected output for the given input?
+    * You can use the examples provided in the prompt, or other examples.
+2. Write unit tests for `get_highest_rated` for the nominal and edge cases you identified in the first step.
+
+*Note: Click the **Run Tests** button to save your tests for instructor feedback. No real tests are actually run again your unit tests.*
+##### !end-question
+
+##### !placeholder
+
+```py
+# example input 2:
+# example output 2:
+
+# example input 2:
+# example output 2:
+
+def test_nominal_case():
+    # ^rename with meaninful test name
+    # and complete test implementation below
+    pass
+    # arrange
+
+    # act
+
+    # assert
+
+def test_edge_case():
+    pass
+```
+
+##### !end-placeholder
+
+##### !tests
+
+```py
+import unittest
+
+class TestPython1(unittest.TestCase):
+  def test_always_pass(self):
+    self.assertEqual(1,1)
+```
+
+##### !end-tests
+##### !explanation 
+
+Example tests:
+
+```python
+def test_picks_highest_rated_from_list():
+    # arrange
+    restaurants = [
+            {"name": "Grillby's", "rating": 1},
+            {"name": "Crow's Nest", "rating": 5}
+        ]
+
+    # act
+    output = get_highest_rated(restaurants)
+    
+    # assert
+    assert output["name"] == "Crow's Nest"
+    assert output["rating"] == 5
+    assert len(output.keys()) == 2
+
+def test_picks_from_list_of_one():
+    # arrange
+    restaurants = [{"name": "Crow's Nest", "rating": 1}]
+
+    # act
+    output = get_highest_rated(restaurants)
+    
+    # assert
+    assert output["name"] == "Crow's Nest"
+    assert output["rating"] == 1
+    assert len(output.keys()) == 2
+
+def test_returns_none_with_zero_restaurants(self):
+    # arrange
+    restaurants = []
+
+    # act
+    output = get_highest_rated(restaurants)
+    
+    # assert
+    assert output == None
```

Comment:
Nitpick that `is` should be used when check whether an object is `None`. If you're unfamiliar, see [an explanation here](https://adac17seattle.slack.com/archives/C02TRJX2QKA/p1645295944921439?thread_ts=1645245325.576119&cid=C02TRJX2QKA) (second half of that comment is the relevant part).
```suggestion
    assert output is None
```